app/src/main/java/com/skydoves/pokedex/ui/details/DetailViewModel.kt

```diff
@@ -16,45 +16,55 @@
 
 package com.skydoves.pokedex.ui.details
 
-import androidx.annotation.MainThread
 import androidx.databinding.ObservableBoolean
-import androidx.hilt.lifecycle.ViewModelInject
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
 import androidx.lifecycle.asLiveData
-import androidx.lifecycle.switchMap
 import com.skydoves.pokedex.base.LiveCoroutinesViewModel
 import com.skydoves.pokedex.model.PokemonInfo
 import com.skydoves.pokedex.repository.DetailRepository
+import com.squareup.inject.assisted.Assisted
+import com.squareup.inject.assisted.AssistedInject
 import timber.log.Timber
 
-class DetailViewModel @ViewModelInject constructor(
-  private val detailRepository: DetailRepository
+class DetailViewModel @AssistedInject constructor(
+  private val detailRepository: DetailRepository,
+  @Assisted private val pokemonName: String
 ) : LiveCoroutinesViewModel() {
 
-  private var pokemonFetchingLiveData: MutableLiveData<String> = MutableLiveData()
-  val pokemonInfoLiveData: LiveData<PokemonInfo?>
-
   val isLoading: ObservableBoolean = ObservableBoolean(false)
   val toastLiveData: MutableLiveData<String> = MutableLiveData()
+  val pokemonInfoLiveData: LiveData<PokemonInfo?> = launchOnViewModelScope(
+    block = {
+      isLoading.set(true)
+      detailRepository.fetchPokemonInfo(
+        name = pokemonName,
+        onSuccess = { isLoading.set(false) },
+        onError = { toastLiveData.postValue(it) }
+      ).asLiveData()
+    }
```

Comment:
I think it is better that initialize in init block for consistency with the MainViewModel. 

```kotlin
  val pokemonInfoLiveData: LiveData<PokemonInfo?>

  init {
    Timber.d("init DetailViewModel")

    pokemonInfoLiveData = launchOnViewModelScope {
      isLoading.set(true)
      detailRepository.fetchPokemonInfo(
        name = pokemonName,
        onSuccess = { isLoading.set(false) },
        onError = { toastLiveData.postValue(it) }
      ).asLiveData()
    }
  }
```

Reply:
Hi @skydoves thanks for the review, I've changed the code plus fix the `DetailActivityInjectionTest`.